docs/quick-start.rst

```diff
@@ -0,0 +1,308 @@
+    </div>
+
+
+by(...)
+~~~~~~~
+
+This modifier splits the frame into groups by the provided column(s), and then
+applies :raw-html:`<b class="i">i</b>` and :raw-html:`<b class="j">j</b>` within
+each group. This mostly affects aggregator functions such as ``sum()``,
+``min()`` or ``sd()``, but may also apply in other circumstances. For example,
+if :raw-html:`<b class="i">i</b>` is a slice that takes first 5 rows of a frame,
+then in the presence of the ``by()`` modifier it will take the first 5 rows of
+each group.
+
+For example, in order to find the total amount of each product sold, write::
+
+    from datatable import f, by, sum
+    DT = dt.fread("transactions.csv")
+
+    DT[:, sum(f.quantity), by(f.product_id)]
+
+
+sort(...)
+~~~~~~~~~
+
+This modifier controls the order of the rows in the result, much like SQL clause
+``ORDER BY``. If used in conjunction with ``by()``, it will order the rows
+within each group.
+
+
+join(...)
+~~~~~~~~~
+
+As the name suggests, this operator allows you to join another frame to the
+current, equivalent to the SQL ``JOIN`` operator. Currently we support only
+left outer joins.
+
+In order to join frame ``X``, it must be keyed. A keyed frame is conceptually
+similar to a SQL table with a unique primary key. This key may be either a
+single column, or several columns::
+
+    X.key = "id"
+
+Once a frame is keyed, it can be joined to another frame ``DT``, provided that
+``DT`` has the column(s) with the same name(s) as the key in ``X``::
+
+    DT[:, :, join(X)]
+
+This has the semantics of a natural left outer join. The ``X`` frame can be
+considered as a dictionary, where the key column contains the keys, and all
+other columns are the corresponding values. Then during the join each row of
+``DT`` will be matched against the row of ``X`` with the same value of the
+key column, and if there are no such value in ``X``, with an all-NA row.
+
+The columns of the joined frame can be used in expressions using the ``g.``
+prefix, for example::
+
+    DT[:, sum(f.quantity * g.price), join(products)]
+
+.. note:: In the future we will expand the syntax of the join operator to
+          allow other kinds of joins, and also to remove the limitation that
+          only keyed frames can be joined.
+
+
+
+Offloading data
+---------------
+
+Just as our work has started with loading some data into datatable, eventually
+you will want to do the opposite: store or move the data somewhere else. We
+support multiple mechanisms for this.
+
+First, the data can be converted into a pandas DataFrame or into a numpy array
+(obviously, you have to have pandas or numpy libraries installed)::
+
+    DT.to_pandas()
+    DT.to_numpy()
+
+A frame can also be converted into python native data structures: a dictionary,
+keyed by the column names; a list of columns, where each column is itself a
+list of values; or a list of rows, where each row is a tuple of values::
+
+    DT.to_dict()
+    DT.to_list()
+    DT.to_tuples()
```

Comment:
Is there any reason as to why we create a list of tuples and not tuple of tuples?

Reply:
Think about this from the types perspective: there is a type `List[T]` for a list of objects of type `T`; then there is also type `Tuple[T1, T2, ..., Tn]` for a tuple of objects of types `T1`, `T2`, ..., `Tn`. So it is natural to use lists for a sequence of same-typed elements of arbitrary length, and it is natural to use tuples for a fixed-length sequence of items of different types.

Though in python it is possible to put anything into a list, or to create a tuple of any length -- this is much less natural, and goes against principles of other typed languages.

It could be argued though that `DT.to_list()` should really return a tuple of columns instead of a list, but I'm not sure if it's worth it...